homeassistant/components/automation/__init__.py

```diff
@@ -510,19 +504,24 @@ async def async_trigger(self, run_variables, context=None, skip_condition=False)
                 variables = run_variables
             automation_trace.set_variables(variables)
 
+            # Prepare tracing the evaluation of the automation's conditions
+            trace_clear()
+            automation_trace.set_condition_trace(trace_get())
+
             if (
                 not skip_condition
                 and self._cond_func is not None
                 and not self._cond_func(variables)
             ):
                 self._logger.debug(
                     "Conditions not met, aborting automation. Condition summary: %s",
-                    condition_trace_get(),
+                    trace_get(),
                 )
-                automation_trace.set_condition_trace(condition_trace_get())
                 return
-            automation_trace.set_condition_trace(condition_trace_get())
-            condition_trace_clear()
+
+            # Prepare tracing the execution of the automation's actions
+            trace_clear()
```

Comment:
Since this is always get just before `trace_get`, would it make sense to turn it into a parameter `trace_get(clear=True)`

Reply:
Fixed in 5d4f9dc